### DISCUSSION TITLE
Enhancement: Decouple framer-motion from Chakra UI to reduce bundle size

### DISCUSSION THREAD

**Discussion by kevinxh** [chakra-ui/chakra-ui#3369] [Ideas]
# 🚀 Feature request

Hi folks! First of all, I'd like to give an applause for creating/maintaining Chakra UI, it's a well designed component library, and I see a bright future for broader adoption in the React community.

However, I'd like to bring up an issue that stopped us from using this component library, which is the excessive bundle size brought by `framer-motion`. This animation library along has a gzip size of 27.57kb, the combination cost of Chakra (40kb) + emotion (8kb) + framer-motion (28kb) is 76kb, which is a lot for many applications, especially mobile focused web apps.
It would be good if we can have framer-motion as an optional for those projects who don't need animations.
 
## 🧱 Problem Statement / Justification

> Please provide valid reason(s) why this should added to Chakra UI.
>
> If this feaure is related to a problem you've noticed. Mention it as well

<img width="370" alt="Screen Shot 2021-02-11 at 2 40 15 PM" src="https://user-images.githubusercontent.com/10948652/107708420-4f76e780-6c78-11eb-9bd0-cc442154e2c2.png">

<img width="572" alt="Screen Shot 2021-02-11 at 2 49 53 PM" src="https://user-images.githubusercontent.com/10948652/107708482-6ddce300-6c78-11eb-8d11-5130f99a6459.png">

![Screen Shot 2021-02-11 at 2 49 40 PM](https://user-images.githubusercontent.com/10948652/107708593-9664dd00-6c78-11eb-891c-8ffa6ee62827.png)

## ✅ Proposed solution or API

> Please provide code snippets, gists, or links to the ideal design or API

## ↩️ Alternatives

> What alternative solutions have you considered before making this request?

## 📝 Additional Information

> What resources (links, screenshots, etc.) do you have to assist this effort?

**Comment by segunadebayo** [comment]
Thanks for sharing your thoughts @kevinxh.

I remember bringing this up with the team as well and we're looking into how to make this work. I'd love to keep this as a discussion to get ideas from the community on how to achieve this.

**Comment by codetheweb** [comment]
One option might to export them under a different file, i.e.

```ts
import { Fade, ScaleFade, Slide, SlideFade } from "@chakra-ui/react/transitions"
```

and then `/transitions` can be tree-shook (is that a phrase? 😛) out of the bundle. This is similar to how Material UI handles imports.

**Comment by frysztak** [comment]
I have some good news: next release of `framer-motion` [will support tree-shaking under Webpack](https://github.com/framer/motion/pull/1008). I'll prepare a PR for Chakra to utilise `m` component from framer-motion. It won't solve this issue completely, but it will reduce bundle size

**Reply by with-heart** [reply]
Nice work! Thank you @frysztak! Looking forward to your PR.

**Reply by frysztak** [reply]
Here it is: #3488, would you care to take a look?
